assets/src/data/model/validators.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * External imports
+ */
+import { Exception } from '@eventespresso/eejs';
+import { sprintf, __ } from '@eventespresso/i18n';
+import { isArray, isEmpty } from 'lodash';
+
+/**
+ * Validates whether the given key exists in the provided entity object.
+ * This is used when calling code wants an exception to be thrown.
+ *
+ * @param { string } key
+ * @param { Object } entity
+ * @param { string } customMessage
+ * @throws { Exception }  Throws an exception if the provided entity does not
+ * 						  have the given key.
+ */
+export const validateEntityHasKey = ( key, entity, customMessage = '' ) => {
+	const message = customMessage === '' ?
+		sprintf(
+			__( 'The provided entity (%s) does not have the given property (%s)',
+				'event_espresso',
+			),
+			entity,
+			key,
+		) :
+		customMessage;
+	if ( ! entity.hasOwnProperty( key ) ) {
+		throw new Exception( message );
+	}
+};
```

Comment:
except for the extra comma after translation namespace:

```
+				'event_espresso',
```

this all looks fine, but in general, I would suggest not swapping variables around so much as it just uses more more memory, both for the actual computers as well as for the devs that have to keep track of everything while reading code.

So instead of having a `customMessage` variable AND a `message` variable, why not just use `message`, and if an  incoming message is not supplied, then use a default value:

```
export const validateEntityHasKey = ( key, entity, message = '' ) => {
	if ( message === '' ) {
		message = sprintf(
			__(
				'The provided entity (%s) does not have the given property (%s)',
				'event_espresso'
			),
			entity,
			key,
		)
	}
	if ( !entity.hasOwnProperty( key ) ) {
		throw new Exception( message );
	}
};
```

bottom line:  there is no reason to introduce another variable when there is no need to preserve the original value, especially in a simple function like this.

Reply:
> except for the extra comma after translation namespace:

That's to satisfy the eslint rules.

I'm not too picky about the variables because the files are built and that gets taken care of in the built results anyways.  However, to keep things concise I will make the requested change.